src/http/server.rs

```diff
@@ -66,30 +73,103 @@ fn create_allowed_headers(headers: &HeaderMap, allowed: &BTreeSet<String>) -> He
 
   new_headers
 }
+
+fn load_cert(filename: &str) -> Result<Vec<rustls::Certificate>, std::io::Error> {
+  let file = File::open(filename)?;
+  let mut file = BufReader::new(file);
+
+  let certificates = rustls_pemfile::certs(&mut file)?;
+
+  Ok(certificates.into_iter().map(rustls::Certificate).collect())
+}
+
+fn load_private_key(filename: &str) -> Result<PrivateKey> {
+  let file = File::open(filename).map_err(CLIError::from)?;
+  let mut file = BufReader::new(file);
+
+  let keys = rustls_pemfile::read_all(&mut file).map_err(CLIError::from)?;
+
+  if keys.len() != 1 {
+    return Err(CLIError::new("Expected a single private key").into());
+  }
+
+  let key = keys.into_iter().find(|key| {
+    matches!(
+      key,
+      rustls_pemfile::Item::RSAKey(_) | rustls_pemfile::Item::ECKey(_) | rustls_pemfile::Item::PKCS8Key(_)
+    )
+  });
+
+  if let Some(key) = key {
+    log::info!("🔑 Loaded private key");
+
+    Ok(match key {
+      rustls_pemfile::Item::RSAKey(key) => PrivateKey(key),
+      rustls_pemfile::Item::ECKey(key) => PrivateKey(key),
+      rustls_pemfile::Item::PKCS8Key(key) => PrivateKey(key),
+      _ => unreachable!(),
+    })
+  } else {
+    Err(CLIError::new("No private key found").into())
```

Comment:
I think the error message needs to be different here. The key was actually found but doesn't match any of the above options.

Reply:
This method can also be used to read other stuff that is not private key (e.g., `X509Certificate` – a certificate – and `Crl` – a certificate revocation list –), so if did not matched the other condition the key was not found